gateway-ha/src/main/java/io/trino/gateway/ha/handler/RoutingTargetHandler.java

```diff
@@ -81,13 +83,19 @@ public boolean isPathWhiteListed(String path)
 
     private String getBackendFromRoutingGroup(HttpServletRequest request)
     {
+        MultivaluedMap<String, String> headers = new MultivaluedHashMap<>();
+        Enumeration<String> headerNames = request.getHeaderNames();
+        while (headerNames.hasMoreElements()) {
+            String key = headerNames.nextElement();
+            headers.add(key, request.getHeader(key));
```

Comment:
Header can be set multiple times with different values.
This only gets the first value. See `getHeaders(String name)`.